diff -dur dlm-3.99.5.orig/dlm_controld/main.c dlm-3.99.5/dlm_controld/main.c
--- dlm-3.99.5.orig/dlm_controld/main.c	2012-06-21 23:53:56.000000000 +0200
+++ dlm-3.99.5/dlm_controld/main.c	2012-11-02 09:07:56.348243739 +0100
@@ -1592,6 +1592,15 @@
 	rv = sigaction(SIGTERM, &act, NULL);
 	if (rv < 0)
 		return -rv;
+	rv = sigaction(SIGINT, &act, NULL);
+	if (rv < 0)
+		return -rv;
+
+	memset(&act, 0, sizeof(act));
+	act.sa_handler = SIG_IGN;
+	rv = sigaction(SIGHUP, &act, NULL);
+	if (rv < 0)
+		return -rv;
 
 	memset(&act, 0, sizeof(act));
 	act.sa_handler = sigchld_handler;
